### PR TITLE
*: Remove function higherOrder & ExternalModels & FindCluster

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #4640

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by SeaRise — APPROVED** [review]

**Review by gengliqi — APPROVED** [review]

File: dbms/src/Interpreters/ExternalModels.h [hunk 1]
```diff
@@ -1,59 +0,0 @@
...
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-#pragma once
-
-#include <Dictionaries/CatBoostModel.h>
```
Comment (JaySon-Huang) [inline dbms/src/Interpreters/ExternalModels.h:17]: Can we also remove the `CatBoostModel`?

Comment (ywqzzy) [reply dbms/src/Interpreters/ExternalModels.h:17]: > Can we also remove the `CatBoostModel`?

Yes, I have another pr that removed some models.

**Pull request merged**